app/src/main/res/layout/activity_main.xml

```diff
@@ -13,6 +13,7 @@
     <com.phlox.tvwebbrowser.activity.main.view.CursorLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
+        android:keepScreenOn="@id/scKeepScreenOn"
```

Comment:
Nope. This wont work like so. Here you need to pass a boolean variable to `android:keepScreenOn` parameter, but `@id/scKeepScreenOn` is basically the Id of your settings switch in settings layout, it doesn't pass it's value like so.

If you want to change `KeepScreenOn` parameter dynamically you should add a specific flag to activity's window. Like in [this ](https://developer.android.com/training/scheduling/wakelock#screen)example.

Reply:
Thank you very much. Here I was very unsure whether kotlin expects more of a programmatic solution, or if the `@...` syntax reacts to changes.

I am just on the way to head out, but will read through it later and then adjust.

Reply:
It is now changed (hopefully correctly).

Unfortunately I couldn't try it out yet, cause Andoroid Studio shows some incompatibility errors (Win 11), which I have to troubleshoot first. I may be able to try it out using a Win 10 machine in a few days.